build-support/native-image/generate-native-image-for-pants-targets.bash

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env bash
+# Script to generate a platform-specific native-image of the pantsbuild zinc wrapper which works to
+# compile macros, from a pants project.
+
+# Requires `jq` and `go`. `jq` can be downloaded from https://stedolan.github.io/jq/!
+# Works on OSX and Linux -- see the README.md in this directory for more usage info.
+
+# TODO: This will be made automatic in pants via https://github.com/pantsbuild/pants/pull/6893.
+
+set -euxo pipefail
+
+# TODO: Right now, NATIVE_IMAGE_EXTRA_ARGS='-H:IncludeResourceBundles=org.scalactic.ScalacticBundle'
+# is necessary to build any zinc native-image which compiles any code using scalatest. This
+# information should be inferred by the native-image agent, but isn't yet.
+
+# TODO: build off of a more recent graal sha (more recent ones fail to build scalac and more): see
+# https://github.com/oracle/graal/issues/1448.
+
+
+# ARGUMENTS
+
+# $@: Forwarded to a `./pants compile` invocation which runs with reflection tracing enabled.
+# $NATIVE_IMAGE_EXTRA_ARGS: Forwarded to a `native-image` invocation.
+
+
+# CONSTANTS
+
+SCRIPT_DIR="$(dirname "${BASH_SOURCE[0]}")"
+
+# shellcheck source=build-support/native-image/build-zinc-native-image.bash
+source "${SCRIPT_DIR}/build-zinc-native-image.bash"
+
+GENERATED_CONFIG_DIR="$(normalize_path "${GENERATED_CONFIG_DIR:-generated-reflect-config}")"
+GENERATED_CONFIG_JSON_FILE="${GENERATED_CONFIG_DIR}/reflect-config.json"
+GENERATED_BUILD_FILE="${GENERATED_CONFIG_DIR}/BUILD"
+
+
+# FUNCTIONS
+
+function download_buildozer {
+  >&2 go get github.com/bazelbuild/buildtools/buildozer || return "$?"
```

Comment:
If this is the only reason for the `go` dependency, IIRC we have a buildozer checked in to binaries.pantsbuild.org that we could just download :)

Reply:
We could! This was a hack, I used to have a link to `download_binary.sh` in this script but couldn't remember what the pantsbuild/binaries name for buildozer was! I will revise this to download it from our s3.

Reply:
Fixed!